### ISSUE TITLE
How to create a surface like this...

### ISSUE THREAD

**Issue by GimpMaster** [8thwall/xr-unity#58]
I'm curious if you guys have some sample code on creating a surface like the one shown in the screenshot below. It grows organically as you move around.

![screenshot_20180729-110205](https://user-images.githubusercontent.com/7818200/43369047-15cbd1ce-9324-11e8-9ad3-0d489a1e5abe.png)

**Comment by dattanchu** [comment]
I believe it is the https://github.com/8thwall/xr-unity/blob/master/scripts/MeshVisualizer.cs in this repo. Also, very nice charger, would be cool to AR in more RC projects.

**Comment by GimpMaster** [comment]
Thank you. I don't remember if the mesh visualizer had as uniform of a surface display. Do you think this is just a texture places over a polygon with some fading on the edges. I'm always interested on how people make some of the cooler effects.

Yeah I got this charger a while ago. There are some newer ones that I want to get that don't limit the current as much.

**Comment by atomarch** [comment]
To get the "grow" effect as you move around, simply attach XRSurfaceController.cs to a 3D Plane game object and make sure the "Deform Surface" option is checked.  The MeshVisualizer.cs script that @dattanchu mentions will work to show you the shape of the surface, but it looks a bit different.  The one in your screenshot looks like PlaneGrid.mat from Google (with TriGrid.png texture and PlaneGrid.shader shader) :

**Comment by GimpMaster** [comment]
Excellent Thank you....quick question. Can you have multiple XRSurfaceController's in a scene?

**Comment by GimpMaster** [comment]
I got this to work like you said. I put a Mesh Filter, Colider, Rendered onto the XR Surface Script game object. I downloaded the AR core PlaneGrid material, TriGrid texture, and Plane Gride shader. 

I then adjusted the material to point at the correct shader and texture.

I had to adjust the Plane Normal to X, Y, Z, W = 0, 90, 0, 1

After that it worked great and the deform to surface adjusted the mesh filter and I got the effect I wanted.
